app/actions.py

```diff
@@ -92,30 +92,172 @@ def performDeleteRange(self, upperRow, upperCol, lowerRow, lowerCol):
         self.getText(upperRow, upperCol, lowerRow, lowerCol)))
       self.redo()
 
+  def resizeBookmarkList(self):
+    """
+    Repeatedly doubles the length self.bookmarks until
+    its length is at least the number of lines in the document.
+
+    Args:
+      None
+
+    Returns:
+      None
+    """
+    while len(self.bookmarks) < len(self.lines):
+      # Double the length of the array
+      length = len(self.bookmarks)
+      self.bookmarks.extend([None] * max(1, length))
```

Comment:
I'm still looking over this PR, so I'm not sure why this is looking for the next higher power of two value, but I think this line may give the desired number without looping:
```
newBookmarksLength = 1<<(len(self.lines)-1).bit_length()
```

Reply:
Oh wow I didn't know there was such a function. Good to know!